usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateAllOptionalFileNameTest.java

```diff
@@ -0,0 +1,186 @@
+
+            for (int i = 0; i < this.howManyOpt; ++i) {
+                toReturn.add("opt" + i + ".opt");
+            }
+
+            return toReturn;
+        }
+
+        @Override
+        public List<String> getRequiredHeadersForFile(RawFileInfo fileInfo) {
+            return null;
+        }
+
+        @Override
+        public List<String> getOptionalHeadersForFile(RawFileInfo fileInfo) {
+            return null;
+        }
+
+        @Override
+        public RawEntityParser getParserForFile(RawFileInfo file) {
+            return null;
+        }
+
+        @Override
+        public ParsedEntityTypeValidator getValidatorForFile(RawFileInfo file) {
+            return null;
+        }
+    }
+
+    //mock raw file repo
+    private static class MockRawFileRepo implements RawFileRepository {
+
+        private final int howManyReq;
+        private final int homManyOpt;
+
+        public MockRawFileRepo(int howManyReq, int howManyOpt) {
+            this.howManyReq = howManyReq;
+            this.homManyOpt = howManyOpt;
+        }
+
+        @Override
+        public RawFileInfo create(RawFileInfo fileInfo) {
+            return null;
+        }
+
+        @Override
+        public Optional<RawFileInfo> findByName(String filename) {
+            return Optional.empty();
+        }
+
+        @Override
+        public Collection<String> getActualHeadersForFile(RawFileInfo file) {
+            return null;
+        }
+
+        @Override
+        public Set<String> getFilenameAll() {
+            Set<String> toReturn = new HashSet<>(this.howManyReq + this.homManyOpt);
+
+            for (int i = 0; i < this.howManyReq; ++i) {
+                toReturn.add("req" + i + ".req");
+            }
+
+            for (int j = 0; j < this.homManyOpt; ++j) {
+                toReturn.add("opt" + j + ".opt");
+            }
+
+            return toReturn;
+        }
+
+        @Override
+        public Optional<RawEntityProvider> getProviderForFile(RawFileInfo file) {
+            return Optional.empty();
+        }
+    }
+
+    //mock validation result repo
+    private static class MockValidationResultRepo implements ValidationResultRepository {
+        public List<Notice> notices = new ArrayList<>();
+
+        @Override
+        public InfoNotice addNotice(InfoNotice newInfo) {
+            notices.add(newInfo);
+            return null;
+        }
+
+        @Override
+        public WarningNotice addNotice(WarningNotice newWarning) {
+            notices.add(newWarning);
+            return null;
+        }
+
+        @Override
+        public ErrorNotice addNotice(ErrorNotice newError) {
+            notices.add(newError);
+            return null;
+        }
+
+        @Override
+        public Collection<Notice> getAll() {
+            return null;
+        }
+
+        @Override
+        public Notice addNotice(Notice newNotice) {
+            return null;
+        }
+    }
+
+    @Test
```

Comment:
I would put the `@Test` methods at the top of the file, and all other boilerplate code that supports them below it. You'll likely edit to the `@Test` methods far more than the others, and that way you don't have to scroll down to find it.

Reply:
I understand. To ensure consistency the same manipulation should be done in every file.
Will modify.

Reply:
#56, will be done in a separate PR as it will touch all test classes